commit 1db6de7b6a6ee240eb50a13fe1fa1e135d7cb93b
Author: Michel Dänzer <michel.daenzer@amd.com>
Date:   Tue Jan 12 15:42:47 2016 +0900

    glamor: Disable debugging messages other than GL API errors
    
    According to Nicolai Hähnle, the relevant specification says "All
    messages are initially enabled unless their assigned severity is
    DEBUG_SEVERITY_LOW", so we need to explicitly disable the messages we
    don't want to get. Failing that, we were accidentally logging e.g.
    shader stats intended for shader-db.
    
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=93659
    Tested-by: Laurent Carlier <lordheavym@gmail.com>
    Reviewed-by: Emil Velikov <emil.l.velikov@gmail.com>
    Signed-off-by: Michel Dänzer <michel.daenzer@amd.com>
    Reviewed-by: Eric Anholt <eric@anholt.net>

diff --git a/glamor/glamor.c b/glamor/glamor.c
index 116d10c..81aba2d 100644
--- a/glamor/glamor.c
+++ b/glamor/glamor.c
@@ -393,6 +393,9 @@ glamor_setup_debug_output(ScreenPtr screen)
         return;
 
     glEnable(GL_DEBUG_OUTPUT_SYNCHRONOUS);
+    /* Disable debugging messages other than GL API errors */
+    glDebugMessageControl(GL_DONT_CARE, GL_DONT_CARE, GL_DONT_CARE, 0, NULL,
+                          GL_FALSE);
     glDebugMessageControl(GL_DEBUG_SOURCE_API,
                           GL_DEBUG_TYPE_ERROR,
                           GL_DONT_CARE,
